### ISSUE TITLE
[Request] Add a tag for Kodi beta release

### ISSUE THREAD

**Issue by arucard21** [kodi-game/peripheral.joystick#49]
I just installed Kodi v17 beta1 and tried to compile this addon, but I couldn't because of this: `error: ‘PERIPHERAL_CAPABILITIES {aka struct PERIPHERAL_CAPABILITIES}’ has no member named ‘provides_buttonmaps’; did you mean ‘provides_joysticks’?`

I looked into this, and it seems like `provides_buttonmaps` was added to the Peripheral API in version 1.0.21, but Kodi v17 beta1 is only at version 1.0.19. I looked at the other available branches and nothing seemed to specifically match this API version. This is why I thought it might be a good idea if someone could add a tag for each Kodi beta (and maybe alpha) release so it would be easier to test the beta release with this binary addon. If possible, it would be nice to have a tag for this beta1 as well, which I guess would be somewhere after ddb0dcf2faaefca275b0691aa22155f3ea67ee10. 

I'm pretty new to the binary addons, so if I just made a mistake somewhere in compiling the addon, please let me know. I'm trying again now with Kodi master branch so if I've done something incorrectly, I'd appreciate the feedback.

**Comment by garbear** [comment]
@MartijnKaijser What should we do about this? master branch of this add-on builds against 17 beta 2, but not 17 beta 1. Our binary add-ons repo has individual add-on definitions per major version, can we add branches for alpha/betas (rc is stable)?

**Comment by garbear** [comment]
Done. tag is named `17beta1`

**Comment by arucard21** [comment]
Thanks, I just compiled 17beta1 again with Kodi Beta1 installed and everything compiles correctly now. The gamepad configuration wizard is also shown in Kodi. 

Unfortunately, while my gamepad seems to be recognized correctly, I can't configure it with the configuration wizard so I can't use it (directly) in Kodi anymore. I get this error in my kodi.log: 
`DEBUG: Failed to locate add-on for "Wireless Controller"`

Should there be a separate addon that contains the configuration of each type of gamepad? Or is this something for which I should create another issue? For now, I've just mapped my gamepad to some keys in order to use it in Kodi, so if I should just wait until these gamepad configuration addons become available, that's fine too.

**Comment by garbear** [comment]
I'm not sure what the bug is, but I can confirm that commit https://github.com/xbmc/xbmc/commit/ee2dc577a03d86ac61e06bc95aaff1d1eddbf593 from https://github.com/xbmc/xbmc/pull/10309 fixes it

**Comment by garbear** [comment]
Right now, the joystick part of the Peripheral API has two parts: joystick drivers and button maps. While many add-ons will provide joystick drivers ([Steam Controller](https://github.com/kodi-game/peripheral.steamcontroller), [X-Arcade Tankstick](https://github.com/kodi-game/peripheral.xarcade)), I intend for only peripheral.joystick to provide button maps.

This means that Steam Controller and X-Arcade Tankstick add-ons can focus on the driver part and leave the button mapping to another add-on.

**Comment by arucard21** [comment]
Thanks a lot for the help, that did indeed fix the problem. Hope it gets merged soon.

If anyone else has this problem, I did the following to add the fix from garbear/xbmc@ee2dc577a03d86ac61e06bc95aaff1d1eddbf593 to the 17.0b1-Krypton code (since I couldn't cherry-pick it from xbmc/xbmc@ee2dc577a03d86ac61e06bc95aaff1d1eddbf593):
```
git remote add garbear git@github.com:garbear/xbmc.git
git pull garbear
git cherry-pick ee2dc577a03d86ac61e06bc95aaff1d1eddbf593
```

**Comment by arucard21** [comment]
Since the tag has been added already (and it works, using the additional commit), I think this issue can be closed.
